dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -129,5 +132,32 @@ private async Task<string> RenderAsync(List<Block> blocks, Kernel kernel, Kernel
         return resultString;
     }
 
+    private void AddMissingInputVariables()
+    {
+        // Distinct variables from the prompt template config
+        var inputVariableNames = new HashSet<string>(this._promptConfig.InputVariables.Select(p => p.Name).ToList(), StringComparer.OrdinalIgnoreCase);
+
+        // Variables from variable blocks e.g. "{{$a}}"
+        var variableNames = this._blocks.Value.Where(block => block.Type == BlockTypes.Variable).Select(block => ((VarBlock)block).Name).ToList();
```

Comment:
Is there still benefit to blocks being Lazy, or should the `Lazy<>` be removed at this point? Doing the parsing promptly also means errors in the prompt will be surfaced out of the call passing in the prompt, presumably leading to better diagnosability?